File: reverse-words-in-a-string/reverse-words-in-a-string.go
func reverseWords(s string) string {
    var results []string
    words := strings.Split(s, " ")
    
    for i := 0; i < len(words) / 2; i++ {
        idx := (len(words) - i) - 1
        words[i], words[idx] = words[idx], words[i]
    }
    
    for i := 0; i < len(words); i++ {
        if strings.TrimSpace(words[i]) != "" {
            results = append(results, words[i])
        }
    }
    
    return strings.TrimSpace(strings.Join(results, " "))
}

File: reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let result = ""
    const words = s.split(" ").filter(Boolean)
    
    for(let i = words.length - 1; i >= 0; i--) {
        if (i !== 0) {
            result += words[i] + " "
        } else {
            result += words[i]
        }
    }
    
    return result
};